Marlin/Marlin_main.cpp

```diff
@@ -6616,9 +6788,11 @@ inline void gcode_M109() {
 
     #if ENABLED(PRINTER_EVENT_LEDS)
       // Gradually change LED strip from violet to red as nozzle heats up
-      if (!wants_to_cool) {
-        const uint8_t blue = map(constrain(temp, start_temp, target_temp), start_temp, target_temp, 255, 0);
-        if (blue != old_blue) set_led_color(255, 0, (old_blue = blue));
+      if (wait_for_heatup && !wants_to_cool) {
+        uint8_t blue = map(constrain(temp, start_temp, target_temp), start_temp, target_temp, 255, 0);
+        if (blue == old_blue) set_rgb_color(255, 0, 255, 0, 0, 1);   //Purple to start
+        if (blue != old_blue) set_rgb_color(255, 0, blue, 0, 0, 1);  //Start transitioning to Red
```

Comment:
These lines resurrect the bad logic I pointed out earlier. Please just extend add the new parameters to the existing `set_led_color` line. Let's preserve the `set_led_color` function name and keep it as the main setter.